frontend/javascripts/oxalis/view/left-border-tabs/histogram_view.js

```diff
@@ -185,6 +194,98 @@ class Histogram extends React.PureComponent<HistogramProps, HistogramState> {
+    const state = Store.getState();
+
+    const [curX, curY, curZ] = Dimensions.transDim(
+      Dimensions.roundCoordinate(getPosition(state.flycam)),
+      viewport,
+    );
+    const [halfViewportExtentX, halfViewportExtentY] = getHalfViewportExtentsFromState(
+      state,
+      viewport,
+    );
+
+    const min = Dimensions.transDim(
+      V3.sub([curX, curY, curZ], [halfViewportExtentX, halfViewportExtentY, 0]),
+      viewport,
+    );
+    const max = Dimensions.transDim(
+      V3.add([curX, curY, curZ], [halfViewportExtentX, halfViewportExtentY, 1]),
+      viewport,
+    );
+
+    const resolutionIndex = getRequestLogZoomStep(state);
+    const cuboid = await api.data.getDataFor2DBoundingBox(layerName, { min, max }, resolutionIndex);
+    return cuboid;
+  };
+
+  getClippingValues = async (layerName: string, threshold: number = 0.05) => {
+    const { elementClass } = getLayerByName(Store.getState().dataset, layerName);
+    const [TypedArrayClass] = getConstructorForElementClass(elementClass);
+
+    const [cuboidXY, cuboidXZ, cuboidYZ] = await Promise.all([
+      this.getDataForViewport(OrthoViews.PLANE_XY, layerName),
+      this.getDataForViewport(OrthoViews.PLANE_XZ, layerName),
+      this.getDataForViewport(OrthoViews.PLANE_YZ, layerName),
+    ]);
+    const dataForAllViewports = new TypedArrayClass(
+      cuboidXY.length + cuboidXZ.length + cuboidYZ.length,
+    );
+
+    dataForAllViewports.set(cuboidXY);
+    dataForAllViewports.set(cuboidXZ, cuboidXY.length);
+    dataForAllViewports.set(cuboidYZ, cuboidXY.length + cuboidXZ.length);
+
+    const valueCounts = {};
+    let maxCount = 0;
+    for (let i = 0; i < dataForAllViewports.length; i++) {
+      if (dataForAllViewports[i] !== 0 && dataForAllViewports[i] in valueCounts) {
+        valueCounts[dataForAllViewports[i]] += 1;
+        if (maxCount < valueCounts[dataForAllViewports[i]]) {
+          maxCount = valueCounts[dataForAllViewports[i]];
+        }
+      } else {
+        valueCounts[dataForAllViewports[i]] = 1;
+      }
+    }
+
+    let lowClip = -1;
+    let highClip = -1;
+    const threshValue = (threshold * maxCount).toFixed();
+    for (let i = 0; i < dataForAllViewports.length; i++) {
+      if (valueCounts[dataForAllViewports[i]] > threshValue) {
+        if (dataForAllViewports[i] !== 0) {
+          if (dataForAllViewports[i] < lowClip || lowClip === -1) {
+            lowClip = dataForAllViewports[i];
+          }
+          if (dataForAllViewports[i] > highClip || highClip === -1) {
+            highClip = dataForAllViewports[i];
+          }
+        }
+      }
+    }
+    return [lowClip, highClip];
+  };
+
+  clipHistogram = async (isInEditMode: boolean, layerName: string) => {
+    const [lowClip, highClip] = await this.getClippingValues(layerName);
+    if (lowClip === -1 || highClip === -1) {
+      Toast.warning(
+        "Clipping the histogram failed. The data did not contain any brightness values greater than 0.",
```

Comment:
I'd try to avoid negative words, such as "failed" in the UI, when possible. My suggestion:

```suggestion
        "The histogram could not be clipped, because the data did not contain any brightness values greater than 0.",
```

Reply:
I think, you forgot to apply my suggestion or didn't push it yet? :see_no_evil:

Reply:
Indeed 😅 I pushed it now!